sqlmesh/utils/date.py

```diff
@@ -277,3 +278,14 @@ def time_like_to_str(time_like: TimeLike) -> str:
     if is_date(time_like):
         return to_ds(time_like)
     return to_datetime(time_like).isoformat()
+
+
+def to_end_date(
+    end_and_units: t.Union[t.Tuple[int, IntervalUnit], t.List[t.Tuple[int, IntervalUnit]]]
```

Comment:
Shouldn't the 2nd Union item be `t.Tuple[t.List[int], IntervalUnit]`? Are we expecting timestamps with different units here? Why even support this version? It seems like the only place we use it is in `get_latest`. Do we even need this helper function? Seems very specific to just that one scenario.

Reply:
I agree. This is fairly old code that I forgot to check again after recent refactor.